library/rhel7_020840.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python
+
+# Written By - Jamie Whetsell - glitch@redhat.com
+
+##############################################################################
+
+import grp, pwd, spwd
+import os, sys
+from pwd import getpwuid
+
+def log(msg):
+    logifle = open('/tmp/aaaaaa', 'a')
+    logifle.write(msg)
+    logifle.write("\n")
+    logifle.write("\n")
+    logifle.close()
+
+def init_permissions(check):
+  # get users
+  # loop over users => get home
+  # get . files in home and check owner for user or root
+  changed=False
+  for p in pwd.getpwall():
+    user = p[0]
+    group = p[4]
+    home = p[5]
+    can_login = len(spwd.getspnam(user)[1]) > 5
+    log("--------------  " + check)
+
+    if can_login:
+      # get a list of . files for the user dir
+      dirs = os.listdir( home )
+      for file in dirs:
+        if file.startswith("."):
+          if user != getpwuid(os.stat(home + "/" + file).st_uid).pw_name:
+            if check == True:
+              changed=False
+              #return dict(failed=True, msg="File - {} is not owned by {}".format(home + "/" + file,user))
+            else:
+              uid = pwd.getpwnam(user).pw_uid
+              gid = grp.getgrnam(group).gr_gid
+              os.chown(home + "/" + file, uid, gid)
+              changed=True
+
+  return changed
+
+def main():
+#    print init_permissions(False)
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            check = dict(default=False,required=False),
+        ),
+    )
+
+    changed = False
+
+    try:
+      changed = init_permissions(module.params.get('check'))
+    except Exception as ex:
+        module.fail_json(msg=str(ex))
+
+    res_args = dict(
+        changed = changed, msg = "OK"
+    )
+
+    module.exit_json(**res_args)
```

Comment:
It might makes sense to return the ouput to return its output inside ansible_facts, which is the [recommended way](http://docs.ansible.com/ansible/developing_modules.html#module-provided-facts) to create custom modules that gather facts.